tests/unit/test_charm.py

```diff
@@ -275,3 +275,63 @@ def test_unused_unit_sets_waiting_status_on_consumer_joined(self):
             self.harness.model.unit.status,
             WaitingStatus("inactive unit"),
         )
+
+    def test_alert_rules(self):
+        self.harness.set_leader(True)
+        prom_rel_id = self.harness.add_relation("metrics-endpoint", "prometheus-k8s")
+        workload_rel_id = self.harness.add_relation("configurable-scrape-jobs", "cassandra-k8s")
+        self.harness.add_relation_unit(workload_rel_id, "cassandra-k8s/0")
+        self.harness.update_relation_data(
+            workload_rel_id,
+            "cassandra-k8s",
+            {
+                "scrape_jobs": json.dumps(
+                    [{"metrics_path": "/metrics", "static_configs": [{"targets": ["*:9500"]}]}]
+                ),
+                "alert_rules": json.dumps(
+                    {
+                        "groups": [
```

Comment:
Given this entire section is identical on both sides of the relation, why not reuse via a variable?

Reply:
Done.